x-pack/plugins/security_solution/cypress/e2e/detection_alerts/detection_page_filters.cy.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getNewRule } from '../../objects/rule';
+import {
+  CONTROL_FRAMES,
+  DETECTION_PAGE_FILTER_GROUP_CONTEXT_MENU,
+  DETECTION_PAGE_FILTER_GROUP_RESET_BUTTON,
+  OPTION_LIST_LABELS,
+  OPTION_LIST_VALUES,
+  OPTION_SELECTABLE,
+} from '../../screens/common/filter_group';
+import { createCustomRuleEnabled } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import { login, visit } from '../../tasks/login';
+import { ALERTS_URL } from '../../urls/navigation';
+import { APP_ID, DEFAULT_DETECTION_PAGE_FILTERS } from '../../../common/constants';
+import { formatPageFilterSearchParam } from '../../../common/utils/format_page_filter_search_param';
+import {
+  markAcknowledgedFirstAlert,
+  selectCountTable,
+  waitForAlerts,
+  waitForPageFilters,
+} from '../../tasks/alerts';
+import { ALERTS_COUNT } from '../../screens/alerts';
+import { navigateFromHeaderTo } from '../../tasks/security_header';
+import { ALERTS, CASES } from '../../screens/security_header';
+
+const assertFilterControlsWithFilterObject = (filterObject = DEFAULT_DETECTION_PAGE_FILTERS) => {
+  cy.log(JSON.stringify({ filterObject }));
+
+  cy.get(CONTROL_FRAMES).should((sub) => {
+    expect(sub.length).eq(4);
+  });
+
+  cy.get(OPTION_LIST_LABELS).should((sub) => {
+    filterObject.forEach((filter, idx) => {
+      expect(sub.eq(idx).text()).eq(filter.title);
+    });
+  });
+
+  cy.get(OPTION_LIST_VALUES).should((sub) => {
+    filterObject.forEach((filter, idx) => {
+      expect(sub.eq(idx).text().replace(',', '')).eq(
+        filter.selectedOptions && filter.selectedOptions.length > 0
+          ? filter.selectedOptions.join('')
+          : 'Any'
+      );
+    });
+  });
+};
+
+const resetFilters = () => {
```

Comment:
Why this is not extracted to a reusable method?

Reply:
@MadameSheema .. corrected ✅